tests/rsocket/conftest.py

```diff
@@ -8,18 +11,31 @@
 from rsocket.rsocket_client import RSocketClient
 from rsocket.rsocket_server import RSocketServer
 
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.fixture
+async def lazy_pipe(unused_tcp_port):
+    yield functools.partial(pipe_factory, unused_tcp_port)
+
 
 @pytest.fixture
-def pipe(unused_tcp_port, event_loop):
+async def pipe(unused_tcp_port, event_loop):
+    async with pipe_factory(unused_tcp_port) as components:
+        yield components
+
+
+@asynccontextmanager
+async def pipe_factory(unused_tcp_port, client_arguments=None, server_arguments=None):
     def session(reader, writer):
         nonlocal server
-        server = RSocketServer(reader, writer)
+        server = RSocketServer(reader, writer, **(server_arguments or {}))
```

Comment:
should not server startup be semantically equal to RSocketClient startup?

Reply:
sounds right. i'll look at it again.

Reply:
This code only specifies the ability to pass different parameters to the client and the server in the tests
The actual RSocketClient and RSocketServer are both subclasses of RSocket.

currently the additional in the RSocketClient init is the parameters required for the setup frame (encodings and keepalive related), and as the server doesn't send one i didn't think they should be there (for now?). since the client doesn't provide a lease publisher that isn't there either.

Reply:
as per discussion with @OlegDokuka on slack will rewrite lease

Reply:
pushed fix